### PR TITLE
First pass at level1 clazy checks for qt

### PR DESCRIPTION
**Pull request by mjcarroll**
Cleaning up `clazy` warnings to make sure we are using Qt correctly.

### CODE REVIEW

File: src/MainWindow_TEST.cc [hunk 1]
```diff
@@ -342,7 +342,7 @@ TEST(MainWindowTest,
 
   // Access window after it's open
   bool closed{false};
-  QTimer::singleShot(300, [&closed]
+  QTimer::singleShot(300, App(), [&closed]
   {
     auto win = App()->findChild<MainWindow *>();
```
Comment (jennuine) [inline src/MainWindow_TEST.cc:347]: should this be:
```suggestion
  auto *app = App();
  QTimer::singleShot(300, app, [&closed]
  {
    auto *win = app->findChild<MainWindow *>();
```
?

File: src/MainWindow_TEST.cc [hunk 2]
```diff
@@ -519,7 +519,10 @@ void FindExitDialogButtons(
   ASSERT_NE(nullptr, dialog);
 
   QObject *buttonBox{nullptr};
-  for (const auto& c : dialog->findChildren<QObject *>())
+  // Qt considers range-based for loops over temporary objects
+  // potentially dangerous, so explicitly create a container.
+  const auto children = dialog->findChildren<QObject *>();
```
Comment (jennuine) [inline src/MainWindow_TEST.cc:524]:
```suggestion
  const auto *children = dialog->findChildren<QObject *>();
```

**Review by jennuine — APPROVED** [review]

File: src/MainWindow_TEST.cc (hunk 1 above)
Comment (mjcarroll) [reply src/MainWindow_TEST.cc]: Yes, I went ahead an addressed the rest of the places where we do this.

File: src/MainWindow_TEST.cc (hunk 2 above)
Comment (mjcarroll) [reply src/MainWindow_TEST.cc:524]: [273fac0](https://github.com/gazebosim/gz-gui/pull/584/commits/273fac06c0698033a9ee5db13eb296e791139532)

**Pull request merged**